src/lib/components/gallery-pagination.jsx

```diff
@@ -4,24 +4,43 @@ import PropTypes from "prop-types"
 // hooks
 import { useGallery } from "../hooks/use-gallery"
 
-// components
-import { GalleryPaginationItem } from "./gallery-pagination-item"
-
 // utils
 import classnames from "../utils/classnames"
 
+/***
+ *
+ * Pagination Item
+ * ---
+ * @typedef {element} renderPaginationItem - Expects a component of GallerPaginationItem
+ * @returns {index, active, activeIndex, item}
+ *
+ * @property {number} index - The index of the current pagination item being iterated over.
+ * @property {boolean} active - Whether the current pagination item being iterated over corresponds to the active gallery item.
+ * @property {number} activeIndex - The index of the currently active gallery item.
+ * @property {object} item - The current pagination item being iterated over, as defined by the Array fed to the `<Gallery>` component's `items` prop.
+ */
+
+/**
+ * Displays a greeting to the user.
+ * @param {Props} renderPaginationItem
+ * @param {string} className
```

Comment:
Let's update per JSDoc convention here, something like the following. Their docs have a lot of nice examples for all the options https://jsdoc.app/tags-param.html

```suggestion
 * @param {object} props
 * @param {prop type} props.propName - description of this prop
 * @param {prop type} [props.optionalPropName] - description of this prop
 * @param {prop type} [props.optionalWithDefault=Hello] - description of this prop
```

Reply:
Thanks for the documentation @andrewrubin I didn't know how to document this properly. Please let me know what you think.

Reply:
It's great, thanks for updating that one